src/tests/test_pubkey_pqc.h

```diff
@@ -77,47 +90,54 @@ class PK_PQC_KEM_KAT_Test : public PK_Test {
+         auto rng_keygen = rng_for_keygen(params, *ctr_drbg);
+         auto rng_encaps = rng_for_encapsulation(params, *ctr_drbg);
 
          // Key Generation
-         auto sk = Private_Key(rng_keygen, d.mode());
-         result.test_is_eq("Generated private key", d.map_value(sk.raw_private_key_bits()), vars.get_req_bin("SK"));
-         result.confirm("All prepared random bits used for key generation", rng_keygen.empty());
+         auto sk = Botan::create_private_key(algo_name(), rng_keygen, params);
+         result.test_is_eq("Generated private key",
+                           map_value(params, sk->raw_private_key_bits(), VarType::PrivateKey),
+                           vars.get_req_bin("SK"));
+         inspect_rng_after_keygen(params, rng_keygen, result);
 
          // Algorithm properties
-         result.test_eq("algorithm name", sk.algo_name(), algo_name());
-         result.confirm("supported operation", sk.supports_operation(Botan::PublicKeyOperation::KeyEncapsulation));
-         result.test_gte("Key has reasonable estimated strength (lower)", sk.estimated_strength(), 64);
-         result.test_lt("Key has reasonable estimated strength (upper)", sk.estimated_strength(), 512);
+         result.confirm("Supported operation KeyEncapsulation",
+                        sk->supports_operation(Botan::PublicKeyOperation::KeyEncapsulation));
+         result.test_gte("Key has reasonable estimated strength (lower)", sk->estimated_strength(), 64);
+         result.test_lt("Key has reasonable estimated strength (upper)", sk->estimated_strength(), 512);
 
          // Extract Public Key
-         auto pk = sk.public_key();
-         result.test_is_eq("Generated public key", d.map_value(pk->public_key_bits()), vars.get_req_bin("PK"));
+         auto pk = sk->public_key();
+         result.test_is_eq("Generated public key",
+                           map_value(params, pk->public_key_bits(), VarType::PublicKey),
+                           vars.get_req_bin("PK"));
 
          // Serialize/Deserialize the Public Key
-         auto pk2 = Public_Key(pk->public_key_bits(), d.mode());
+         auto pk2 = Botan::load_public_key(pk->algorithm_identifier(), pk->public_key_bits());
 
          // Encapsulation
-         auto enc = Botan::PK_KEM_Encryptor(pk2, "Raw");
+         auto enc = Botan::PK_KEM_Encryptor(*pk2, "Raw");
          const auto encaped = enc.encrypt(rng_encaps, 0 /* no KDF */);
-         result.test_is_eq("Shared Secret", encaped.shared_key(), Botan::lock(vars.get_req_bin("SS")));
-         result.test_is_eq("Ciphertext", d.map_value(encaped.encapsulated_shared_key()), vars.get_req_bin("CT"));
-         result.confirm("All prepared random bits used for encapsulation", rng_encaps.empty());
+         result.test_is_eq("Shared Secret",
+                           encaped.shared_key(),
+                           Botan::lock(map_value(params, vars.get_req_bin("SS"), VarType::SharedSecret)));
```

Comment:
I think the mapping should happen on the algorithm's result not the KAT input, no?

```suggestion
                           map_value(params, encaped.shared_key(), VarType::SharedSecret),
                           Botan::lock(vars.get_req_bin("SS"));
```

Reply:
Good catch 😅